src/main/java/de/dennisguse/opentracks/util/EGM2008Utils.java

```diff
@@ -65,7 +65,7 @@ static Indices getIndices(Location location) {
         if (location.getLongitude() >= 0) {
             longitude = location.getLongitude();
         } else {
-            longitude = 180 + Math.abs(location.getLongitude());
+            longitude = 360 + location.getLongitude();
```

Comment:
So here you are converting -180 to 0 into 180 to 360.   This is basically normalizing the value into [0,360) east longtitude.   I missed earlier that it's happening only for negative longitude.